mongorestore/mongorestore_test.go

```diff
@@ -487,3 +487,110 @@ func TestKnownCollections(t *testing.T) {
 		})
 	})
 }
+
+
+func TestFixHashedIndexes(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	type indexRes struct {
+		Key bson.D
+	}
+
+	Convey("Test MongoRestore with hashed indexes and --fixHashedIndexes", t, func() {
+		args := []string{
+			FixDottedHashedIndexesOption,
+		}
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		session, _ = restore.SessionProvider.GetSession()
+		db := session.Database("testdata")
+
+		defer func() {
+			db.Collection("hashedIndexes").Drop(nil)
+		}()
+
+		Convey("Once collection foo has been restored, it should exist in restore.knownCollections", func() {
+			restore.TargetDirectory = "testdata/hashedIndexes.bson"
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+
+			var namespaceExistsInCache bool
+			if cols, ok := restore.knownCollections["testdata"]; ok {
+				for _, collName := range cols {
+					if collName == "hashedIndexes" {
+						namespaceExistsInCache = true
+					}
+				}
+			}
+			So(namespaceExistsInCache, ShouldBeTrue)
```

Comment:
I don't think checking the cache is necessary here? `result := restore.Restore(); So(result.Err, ShouldBeNil)` should be enough to check the restore succeeded and you can go on to checking it worked correctly.

Reply:
I just copied the previous test :)

Reply:
I'll remove this. It's pretty extraneous to what's actually being tested here.